common/deserialization/subtitles/subtitles.cpp

```diff
@@ -45,7 +45,9 @@ bool write_subtitle_db_to_files(const GameSubtitleDB& db) {
         }
         file_contents += "\n";
         if (scene_info.m_lines.empty()) {
-          file_contents += fmt::format("  ()\n");
+          // prevent creating empty lines, which causes crash on startup
+          file_contents +=
+              fmt::format("  (0 :offscreen \"ERROR!\" \"THIS SCENE WAS EMPTY ON SAVE!\")\n ");
```

Comment:
the line wasn't ever removed, it was () when I first touched this and the game would exit when it encountered the empty list on startup. I had it as (0) to prevent the crash, I can change it back to this, I honestly just 1.didnt wanna go trying to figure out how to get the game to handle an empty list gracefully and 2. wanted to differentiate this case from the lone (0) I had it generate in other, expected cases (I did end up catching something this way, I believe)

Reply:
So what im trying to say is that the original code printed:
- `()\n`
When it should have printed
- `'()\n`

This may be the only change needed (the proper way to write an empty list)

Reply:
Sorry for misunderstanding you there, I didn't read carefully enough

Tried `'()` just now; it throws [`Error: Each entry must start with a timestamp (number)`](https://github.com/open-goal/jak-project/blob/2d595c1ac0bf128c91b638728d98e79c42f50883/common/serialization/subtitles/subtitles.cpp#L290) at `parse_subtitle` rather than `Each entry must be a list` as happens with `()`.

Writing an empty scene with a single `(0)` would mean hints will always have something to edit, but adding a new hint in the editor already does this, so probably not necessary to handle an empty entry like that. For now, simply skipping adding the entry to the scene seems to work best as it'll build / run the game fine, but while `if (entry.is_empty_list())` fails to catch `'()`, it does catch `()` -- so either i need to use a different check here, or the print should go back to how it was.

EDIT: hit the wrong button

Reply:
I'm alright with you using your solution, I just wondered if it could be simplified.

It sounds like the current parser isn't handling empty lists properly -- so that's something we'll want to fix at some point in the future, or defer it to when OpenGOAL has a more proper way of serializing/deserializing data -- instead of having to make bespoke parsers/emitters for everything.

Reply:
Makes more sense to me, but doesn't feel like my call (plus i wouldn't know how to go about converting the files from the old to the new pattern when this transition occurs, lol). Either way, the check's there now and prevents the game from just exiting on startup (though i print an error, it gets buried, which isn't totally ideal)

Reply:
`()` is not a valid subtitle line. This is deliberate. It doesn't make sense not to specify a timestamp, since it is not possible for a subtitle line not to have a timestamp.

Reply:
The reason `()` and `'()` have different behavior is because the former is a special "empty list" object while the latter is a pair like you would expect, and I think its value is actually `(quote ())`. Keep in mind `'` is just syntactic sugar.

The difference between your two examples I think is superfluous. The `cdr` of
```lisp
("sidekick-human-intro-sequence-b"
  (937)
  (1027))
```
is
```lisp
((937) (1027))
```

which isn't very different from the `cadr` of your second example, except now you have to check if that is a list first:
```lisp
("sidekick-human-intro-sequence-b"
  ((937)
   (1027)))
```
`cdr`:
```lisp
(((937) (1027)))
```
`car`:
```lisp
((937) (1027))
```

Reply:
Having just
```lisp
("sidekick-human-intro-sequence-b")
```
shouldn't cause issues either, since the `cdr` of that will be the empty list which should be handled correctly by `for_each_in_list`.

Reply:
indeed,
```lisp
(""sidekick-human-intro-sequence-b"
  )
```
is what we end up with if we generate a subtitle that contains only an empty list, and it is parsed just fine. if you're saying it's desirable to throw an exception in the case of a subtitle containing an empty list, then I can change this PR to reflect that